src/routes/mods/[slug]/+page.svelte

```diff
@@ -0,0 +1,194 @@
+    import { getPackageInfo, isInstalledForVersion, getDeplist } from '$lib/api/contentdb';
+    import { downloadAndUnzip } from '$lib/api/download';
+    //import { openGame } from '$lib/shell';
+
+    import { onMount } from 'svelte';
+  	import { writable } from 'svelte/store';
+	
+    let packageInfo = false;
+    let isInstalled = false;
+    let isSupported = false; //TODO: figure out a way to do this...
+
+	let installing = false;
+	let currentinstalling = writable("");
+	let progress = writable(0);
+	let modstoinstall = [];
+	
+    let [ author, pack ] = slug.split('@');
+
+    onMount(updatePackage);
+	page.subscribe(updatePackage);
+
+	async function updatePackage() {
+		slug = $page.params.slug;
+		let slugParts = slug.split('@');
+        packageInfo = await getPackageInfo(slugParts[0], slugParts[1]);
+		//isInstalled = await isInstalledForVersion(slugParts[1], 'mods', $selectedVersion.name);
+	};
+
+    selectedVersion.subscribe(async(val) => {
+        if (!val.installed) {
+			isSupported = false;
+			return;
+		}
+
+        isInstalled = await isInstalledForVersion(pack, 'mods', val.name);
+		for (let i in packageInfo.releasesdata){
+			let release = packageInfo.releasesdata[i];
+			if (release.min_minetest_version){
+				isSupported = release.min_minetest_version.protocol_version >= val.branch.protocolVersion;
+				if (isSupported && release.max_minetest_version) {
+					isSupported = release.max_minetest_version.protocol_version <= val.branch.protocolVersion;
+				}
+			} else if (release.max_minetest_version) {
+				isSupported = release.max_minetest_version.protocol_version <= val.branch.protocolVersion;
+			}
+			if (isSupported) break;
+		}
+	});
+
+    async function install(version = '5.6.0') {
+		if (installing) return;
+		modstoinstall = await getDeplist(packageInfo, version);
+		modstoinstall.push(packageInfo);
+		
+        installing = true;
+		progress.set(0);
+		let installedprogress = 0;
+		for (let i in modstoinstall) {
+			let installpackage = modstoinstall[i];
+			progress.update(n => n + 1);
+			currentinstalling.set(installpackage.name);
+			try {
+				await downloadAndUnzip(installpackage.url, `/versions/${version}/mods/${installpackage.name}`);
+				installedprogress += 1;
+			}
+			catch (err) {
+				console.log(err);
+			}
+		}
+        installing = false;
+		isInstalled = installedprogress == modstoinstall.length;
+		return isInstalled;
+	}
+
+	async function toggleMod(pack, version = '5.6.0'){
+		if (pack || version && isInstalled){
+			/* 
+			* Steps:
+			* 1. check if world.mt (need worlds thing first) includes 'load_mod_${pack}'
+			* 2. if step1 then get the value of 'load_mod_${pack} = <VALUE>' and set NEWVALUE to !VALUE
+			* 3. else set NEWVALUE to true
+			* 4. write the new value as 'load_mod_${pack} = <NEWVALUE>' into the world.mt file
+			*/
+			alert("Need to implement this!");
+		}
+	}
+
+	async function deleteMod(pack, version = '5.6.0'){
```

Comment:
`deleteMod` is something that we would abstract to a dedicated lib file

Reply:
Same for toggleMod, idk where to put